tests/tests.js

```diff
@@ -167,6 +188,52 @@ var runTests = function runTests(arrayFrom) {
 		t.end();
 	});
 
+	test('works with iterable objects', function (t) {
+		(function () {
+			t.deepEqual(arrayFrom(arguments), [1, 2, 3]);
+		}(1, 2, 3));
+
+		if (Array.prototype.values) {
+			t.deepEqual(arrayFrom([1, 2, 3].values()), [1, 2, 3]);
+		}
+
+		if ('Map' in global && 'values' in Map.prototype) {
+
+			var map = new Map();
+			map.set(1, 2);
+			map.set(3, 4);
+			var mapIterator = map.values();
+
+			if (typeof mapIterator.next === 'function') {
+
+				// Test map iterable
+				t.deepEqual(arrayFrom(mapIterator), [2, 4]);
+				t.deepEqual(arrayFrom(map), [[1, 2], [3, 4]]);
+			}
+		}
+
+		if ('Set' in global && 'values' in Set.prototype) {
```

Comment:
tests should be skipped via the `skip` option rather than if blocks - and the Map and Set tests should be different subtests of the generic "iterable" test.